### PR TITLE
Upgrade versions in GH actions. Fix #2985

### LINKED ISSUE #2985: Bump Zotonic Docker images and Gthub CI
Zotonic `Dockerfile.dev` uses the `erlang:24-alpine`, but `erlang:25-alpine` was released.
Locally all green running tests in `Erlang 25.0`.

I also think the `Postgres` version could be bumped. The `docker-compose.yml` uses the `postgres:11.6-alpine`, but the latest is `postgres:14.3-alpine`.

`Github workflows` also use earlier versions:
* `dialyzer.yml`: otp_version: [22.3]
* `docs.yml`: otp_version: [23]
* `elvis.yml`: otp_version: [23]
* `publish.yml`: otp_version: [24.0.5]
* `test.yml`: otp_version: [22, 23.2.5, 24.0.5]

The `elvis.yml` run elvis by the `0.2.12`, latest is `1.1.0`.

### PR DESCRIPTION
**Pull request by mworrell**
### Description

Upgrade versions in GH actions:

 - Add OTP 25
 - Dialyzer on OTP 24
 - Postgres 14

Upgrade:

 - epgsql to 4.6.0 for OTP24+ compatibility 
 - gen_smtp to 1.2.0

Needs additional changes:

 - [x] gen_smtp: https://github.com/gen-smtp/gen_smtp/issues/316

### Checklist

- [ ] documentation updated
- [x] tests added
- [x] no BC breaks

### CODE REVIEW

**Comment by mworrell** [comment]
Fails because of an unexpected function in epgsql:

https://github.com/zotonic/zotonic/runs/6569638121?check_suite_focus=true#step:6:47

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ jobs:
 
     strategy:
       matrix:
-        otp_version: [22, 23.2.5, 24.0.5]
+        otp_version: [22.3, 23.2.5, 24.3, 25]
```
Comment (vkatsuba) [inline .github/workflows/test.yml:21]:
```suggestion
        otp_version: [22, 23, 24, 25]
```
Use this OTP versions, will be always loaded latest minor version.

**Comment by mworrell** [comment]
Reverted elvis to old version, the elvis workflow needs more work anyway

**Comment by mworrell** [comment]
Also needs upgrading gen_smtp

https://github.com/zotonic/zotonic/runs/6570022953?check_suite_focus=true#step:6:224

**Comment by mworrell** [comment]
Waiting for OTP 25.0.1 to fix this issue.

https://github.com/erlang/otp/issues/6015

**Comment by mworrell** [comment]
The issue in OTP has been fixed, we can rerun with 25.0.1 when it is released.

**Comment by mworrell** [comment]
25.0.1 has been released, but apparently not yet available.

**Pull request merged**